libraries/s3/lib_test/test_async.ml

```diff
@@ -0,0 +1,15 @@
+open Aws_s3_test
+
+module T = TestSuite (struct
+  type 'a m = 'a Async.Deferred.t
+
+  let access_key = Unix.getenv "AWS_ACCESS_KEY_ID"
+
+  let secret_key = Unix.getenv "AWS_SECRET_ACCESS_KEY"
+
+  let token = None
+
+  let run_request = Aws_async.Runtime.run_request ~access_key ~secret_key ?token
+
+  let un_m v = Async.Thread_safe.block_on_async_exn (fun () -> v)
+end)
```

Comment:
A couple of improvements:
1. I moved this logic into the `aws_service_test.ml` files so I could conditionally run the tests in CI (supporting both GitHub Actions and ocaml-ci) See https://github.com/inhabitedtype/ocaml-aws/blob/master/libraries/ec2/lib_test/aws_ec2_test.ml#L183-L200
1. the S3 bindings are broken and need some attention on the deserialisation https://github.com/inhabitedtype/ocaml-aws/issues/93

Could you remove the S3 generated code and move the test logic over as per aws_ec2 example, cheers.

Reply:
Hey @tmcgilchrist -- sorry for re-adding the broken S3 code. I removed the generated S3 test stuff from the branch.

I'm slightly confused about your second request. Do you want me to move the test logic for *other* libraries to be similar to the EC2 example you provided?

Let me know :)

Reply:
Looks like it was just the S3 test that wasn't doing what I wanted. No need to do anything now.